cms/djangoapps/contentstore/management/commands/tests/test_delete_orphans.py

```diff
@@ -38,3 +40,45 @@ def test_delete_orphans_commit(self):
         self.assertFalse(self.store.has_item(self.course.id.make_usage_key('vertical', 'OrphanVert')))
         self.assertFalse(self.store.has_item(self.course.id.make_usage_key('chapter', 'OrphanChapter')))
         self.assertFalse(self.store.has_item(self.course.id.make_usage_key('html', 'OrphanHtml')))
+
+    def test_delete_orphans_published_branch(self):
+        """
+        Tests that if there are orphans only on the published branch,
+        running delete orphans with a course key that specifies
+        the published branch will delete the published orphan
+        """
+        course = CourseFactory.create(default_store=ModuleStoreEnum.Type.split)
```

Comment:
I think this is a split-only scenario: a "published-only" orphan in old mongo will be picked up in draft-preferred. This is for the scenario where an orphan block is present in a published course structure but not the draft course structure.

I'm actually about to close this PR and open a new one against the RC; let's continue the conversation there.